source/grid/grid_generator.cc

```diff
@@ -5360,113 +5375,168 @@ namespace GridGenerator
+                                                     {7, 11, 13, 6},
+                                                     {9, 3, 7, 11},
+                                                     {1, 12, 9, 3},
+                                                     {12, 2, 3, 11},
+                                                     {2, 8, 11, 6},
+                                                     {12, 0, 2, 8},
+                                                     {1, 10, 12, 0}};
+
+            cells.resize(n_cells, CellData<3>());
+
+            for (unsigned int i = 0; i < n_cells; ++i)
               {
-                cells[i].vertices[j]     = rhombi[i][j];
-                cells[i].vertices[j + 4] = rhombi[i][j] + 14;
+                for (unsigned int j = 0; j < 4; ++j)
+                  {
+                    cells[i].vertices[j]     = rhombi[i][j];
+                    cells[i].vertices[j + 4] = rhombi[i][j] + 14;
+                  }
+                cells[i].material_id = 0;
               }
-            cells[i].material_id = 0;
+
+            tria.create_triangulation(vertices, cells, SubCellData());
+            break;
           }
+        case 24:
+        case 48:
+          {
+            // These two meshes are created by first creating a mesh of the
+            // 6-cell/12-cell version, refining globally a few times, and
+            // removing the outer half of the cells until done. It would have
+            // been slightly less code to treat refinement steps recursively
+            // for 192 cells or beyond, but unfortunately we could end up
+            // with the 96 cell case which is not what we want. Thus, we need
+            // to implement a loop manually here.
+            Triangulation<3>   tmp;
+            const unsigned int outer_radius_factor = 1 << n_refinement_steps;
+            hyper_shell(tmp,
+                        p,
+                        inner_radius,
+                        outer_radius_factor * outer_radius -
+                          (outer_radius_factor - 1) * inner_radius,
+                        n / 4);
+            for (unsigned int r = 0; r < n_refinement_steps; ++r)
+              {
+                tmp.refine_global(1);
+                std::set<Triangulation<3>::active_cell_iterator>
+                  cells_to_remove;
+
+                // We remove all cells which do not have exactly four vertices
+                // inside the outer radius. Due to roundoff, we do not test
+                // for "square < outer_radius * outer_radius" but something
+                // more stable.
+                for (const auto &cell : tmp.active_cell_iterators())
+                  {
+                    unsigned int n_vertices_inside = 0;
+                    for (const auto v : GeometryInfo<3>::vertex_indices())
+                      if ((cell->vertex(v) - p).norm_square() <
+                          inner_radius * outer_radius)
```

Comment:
I test whether each cell has four vertices inside `outer_radius`. Actually, I look for something close to `inner_radius`, like `inner_radius * (1+1e-12)`. Rather than writing it like this, I used a number that I know for sure is less than `outer_radius` and a bit more than `inner_radius`. Since `inner_radius` and `outer_radius` must differ enough to give a viable mesh, I thought this is the least amount of code to write. But I admit it is harder to follow than necessary, and thus will write `inner_radius * inner_radius * (1 + 1e-12)` to be more descriptive of what I want.

Reply:
I don't particularly care which way you do the test, as long as I can follow the comment :-)